main/lsp/requests/completion.cc

```diff
@@ -1276,7 +1276,7 @@ unique_ptr<ResponseMessage> CompletionTask::runRequest(LSPTypecheckerDelegate &t
         if (enableTypedFalseCompletionNudges) {
             ENFORCE(fref.exists());
             auto level = fref.data(gs).strictLevel;
-            if (!fref.data(gs).hasParseErrors() && level < core::StrictLevel::True) {
+            if (!fref.data(gs).hasIndexErrors() && level < core::StrictLevel::True) {
```

Comment:
This is one place where the change in behavior could affect things. We should audit this call site and see whether this causes problems for whatever completion.cc was trying to use this bit for.

Reply:
This seems a little not great: it will only show the typed-false nudge when the file would already pass all the checks in the indexer. Maybe it's worth tracking parse errors and indexer errors as two separate flags on the `File`?

Reply:
I added this behavior, with the relationship that parse errors imply indexer errors. This means that we can keep the existing behavior of nudging when the file parses, but skipping the cache when there are any errors added during indexing.

Reply:
If this is the only reason why you started to track hasParseErrors and hasIndexErrors, I think it might be worth not going to that trouble.

The original context for this is that sometimes we would give people a completion item seeming to claim that if they upgraded their file to `# typed: true`, they would get completion results.

But in cases where the entire file failed to parse, or parsed but in a way that dropped a ton of context from the tree, we didn't want to show that completion result for fear of misleading the user.

In this case, I think that if for example you tried to get completion results on `X| &&= 0`, this **also** would have made the parse be `EmptyTree`, and thus Sorbet would see no query responses, and would report this error. As you can see, this is [already a problem on master](https://sorbet.run/#%23%20typed%3A%20false%0A%0AXZY%20%3D%201%0A%0AXY%20%26%26%3D%201):

<img width="471" alt="image" src="https://github.com/user-attachments/assets/7a619e1b-747e-44dd-8b6b-5a00eb10ef7b">

So I actually think that there's probably value in just having this callsite say `hasIndexErrors`.

Reply:
Yep, that was the only reason. I'll back that change out, which I think is nice as we no longer have to think about maintaining the `parse errors => index errors` relationship.